### DISCUSSION TITLE
Cache should be available in main domain.

### DISCUSSION THREAD

**Discussion by rahulradixweb** [TruCopilot/phpfastcache#837] [Q&A] [answered]
Hi 
Forgive me for too many questions.

We want to clear store and clear cache for all subdomains (subdomain1.main.com, subdomain2.main.com ) From
main.com , means cache can be stored by subdomain1 or 2 but it should be stored in a centralised path, currently it creates two different directory for each domain, and clear does only clears of domain from where script is called.

Implementation is like.

```
    Phpfastcache\CacheManager::setDefaultConfig(new Phpfastcache\Config\ConfigurationOption([
        "path" => "/tmpcache/",
        "defaultTtl" => 86400
    ]));
    $InstanceCache = Phpfastcache\CacheManager::getInstance('files');
```

It creates

/tmpcache/subdomain1.main.com/Files
/tmpcache/subdomain2.main.com/Files

And When I clear cache from main.com => it doesn't clear subdomain1.main.com/ cache, but I want to maintain at centralised level.

Can you please guide me with this?

Your support is appriciable.

**Comment by Geolim4** [comment, marked as answer]
Hello,

I asked you to read carefully the DOCS and Wiki for a good reason buddy:

https://github.com/PHPSocialNetwork/phpfastcache/wiki/%5BV4%CB%96%5D-Configuration-Options#file-based-drivers-options-

Set the option `securityKey` to the value of your choice. Therefore Phpfastcache will not rely on `HTTP_HOST`...

**Reply by rahulradixweb** [reply]
I try to set custom subdirectory name it throws an exception

[Phpfastcache\Exceptions\PhpfastcacheInvalidConfigurationException, 0] 
Invalid option(s) for the config Phpfastcache\Config\ConfigurationOption: securityKey 
@line 108 in file /~/vendor/phpfastcache/phpfastcache/lib/Phpfastcache/Config/ConfigurationOption.php

**Comment by Geolim4** [comment]
Please read the documentation, for real.

SecurityKey does not belong to ConfigurationOption class.

Look into the Phpfastcache core.
